cloud/azure/actuators/machine/machineactuator_test.go

```diff
@@ -125,6 +126,104 @@ func TestExistsFailureVMNotExists(t *testing.T) {
 		t.Fatalf("expected machine: %v to not exist", machine.ObjectMeta.Name)
 	}
 }
+func TestUpdateVMNotExists(t *testing.T) {
+	azureServicesClient := mockVMNotExists()
+	params := MachineActuatorParams{Services: &azureServicesClient}
+
+	machineConfig := newMachineProviderConfig()
+	machine := newMachine(t, machineConfig)
+	cluster := newCluster(t)
+
+	actuator, err := NewMachineActuator(params)
+	err = actuator.Update(cluster, machine)
+	if err == nil {
+		t.Fatal("expected error calling Update but got none")
+	}
+}
+func TestUpdateMachineNotExists(t *testing.T) {
+	azureServicesClient := mockVMExists()
+	machineConfig := newMachineProviderConfig()
+	machine := newMachine(t, machineConfig)
+	cluster := newCluster(t)
+
+	params := MachineActuatorParams{Services: &azureServicesClient, V1Alpha1Client: fake.NewSimpleClientset().ClusterV1alpha1()}
+	actuator, err := NewMachineActuator(params)
+	err = actuator.Update(cluster, machine)
+	if err == nil {
+		t.Fatal("expected error calling Update but got none")
+	}
+}
+
+// func TestUpdateNoSpecChange(t *testing.T) {
```

Comment:
Would require mocking `util.GetMachine` method or matching the exact `instance-status` annotation structure for testing.